src/Database/LMDB/Simple/Cursor.hs

```diff
@@ -114,13 +116,20 @@ data PeekPoke k v = PeekPoke {
 -- We perform lower-level cursor operations in @'IO'@, while the @'ReaderT'@
 -- passes along a @'CursorEnv'@ that contains all the information we need to
 -- perform the lower-level cursor operations.
-newtype CursorM k v a = CursorM {unCursorM :: ReaderT (CursorEnv k v) IO a}
+newtype CursorM k v mode a = CursorM {unCursorM :: ReaderT (CursorEnv k v) IO a}
```

Comment:
Maybe: document what `mode` represents, and to what it will be instantiated in practice.

Reply:
Will do. I might also change the representation of modes slightly, or possibly do it in a separate PR

```haskell
data Mode = ReadWrite | ReadOnly
```